spec/unit/type/logical_volume_spec.rb

```diff
@@ -0,0 +1,336 @@
+
+  it 'valid readahead does not raise error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name           => 'bert',
+        :ensure         => :present,
+        :volume_group   => 'ernie',
+        :size           => '10M',
+        :readahead      => '7Auto'
+      )
+    }.to_not raise_error
+  end
+
+  it 'size is minsize raises error if not a boolean' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name              => 'kuzco',
+        :ensure            => :present,
+        :volume_group      => 'kronk',
+        :size              => '10M',
+        :size_is_minsize   => 'pacha',
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter size_is_minsize failed on Logical_volume[kuzco]: size_is_minsize must either be true or false')
+  end
+
+  it 'size is minsize does not raise error if boolean' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name             => 'kuzco',
+        :ensure           => :present,
+        :volume_group     => 'kronk',
+        :size             => '10M',
+        :size_is_minsize  => 'true'
+      )
+    }.to_not raise_error
+  end
+
+  it 'mirror number outside of 0-4 range throws error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name              => 'scooby doo',
+        :ensure            => :present,
+        :volume_group      => 'shaggy',
+        :size              => '10M',
+        :mirror            => '-1'
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter mirror failed on Logical_volume[scooby doo]: -1 is not a valid number of mirror copies. Use 0 to un-mirror or 1-4 to set up mirroring.')
+  end
+
+  it 'mirror within 0-4 range does not throw error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name              => 'fred',
+        :ensure            => :present,
+        :volume_group      => 'daphne',
+        :size              => '10M',
+        :mirror            => '1'
+      )
+    }.to_not raise_error
+  end
+
+  it 'region_size with invalid value throws error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name             => 'fred',
+        :ensure           => :present,
+        :volume_group     => 'daphne',
+        :size             => '10M',
+        :region_size      => 'velma'
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter region_size failed on Logical_volume[fred]: velma is not a valid region size in MB.')
+  end
+
+  it 'region_size set to positive integer does not throw error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name             => 'fred',
+        :ensure           => :present,
+        :volume_group     => 'daphne',
+        :size             => '10M',
+        :region_size      => '910'
```

Comment:
I think you had a `'7+'` value here that should throw an error. Could you do that again?